### PR TITLE
Remove a few warnings on macOS

### PR DESCRIPTION
**Pull request by girving**
1. Remove unused captures in lambdas.
2. Reorder a field initializer list.
3. Remove some unused typedefs.
4. Remove a few unused variables.

Many warnings remain, including many thread safety warnings which are
somewhat alarming.

### CODE REVIEW

**Review by jsimsa — DISMISSED** [review]

**Comment by girving** [comment]
@jsimsa Am I correct that the Windows failure is unrelated?  Not sure about the copybara one.

**Comment by jsimsa** [comment]
Correct, the Windows failure seems unrelated. The copybara one tracks the status of the internal review that should get kicked off by @rthadur shortly.

File: tensorflow/core/kernels/data/stats_utils.cc [hunk 1]
```diff
@@ -14,6 +14,7 @@ limitations under the License.
 ==============================================================================*/
 #include "tensorflow/core/kernels/data/stats_utils.h"
 
+#include "absl/base/attributes.h"
```
Comment (jsimsa) [inline tensorflow/core/kernels/data/stats_utils.cc]: Internally, I am seeing an error that indicates that `tensorflow/core/kernels/data:stats_utils` does not depend on a module exporting `third_party/absl/base/attributes.h`. Could you please fix that? Thanks.

Comment (girving) [reply tensorflow/core/kernels/data/stats_utils.cc]: Should be done.

**Comment by girving** [comment]
@jsimsa What's the next step here?

**Review by jsimsa — APPROVED** [review]

**Comment by girving** [comment]
Thank you!

**Pull request merged**